### PR TITLE
Use preferred icon paths and add appdata

### PR DESCRIPTION
**Pull request by Alexander-Wilms**
With these changes, the build manifest becomes a bit shorter as well:
```
app-id: org.librepcb.LibrePCB
runtime: org.kde.Sdk
runtime-version: 5.11
sdk: org.kde.Sdk
command: librepcb
finish-args:
- --socket=wayland
- --socket=x11
- --share=ipc
- --device=dri
- --share=network # required for downloading libraries
- --filesystem=home # required for creating the workspace, storing libraries etc.
modules:

- name: librepcb
  buildsystem: qmake
  builddir: true
  sources:
  - type: git
    url: https://github.com/LibrePCB/LibrePCB
```

### CODE REVIEW

**Comment by Alexander-Wilms** [comment]
I'm not sure why this fails
```
../../../apps/librepcb/main.cpp:124:5: error: ‘setDesktopFileName’ is not a member of ‘librepcb::Application’
     Application::setDesktopFileName("org.librepcb.LibrePCB");
```
My local Flatpak build succeeds. Maybe it due to the different Qt versions (5.9 vs. 5.11)?

**Comment by ubruhin** [comment]
Building the AppImage now fails because the *.desktop file was renamed. Could you update the filenames used in `ci/build.sh`?

I hope all these changes do not break something with older distributions. Generally I try to be compatible with the oldest still supported Ubuntu version (14.04 at this time). I'll try if everything still works on 14.04 before merging this...

File: apps/librepcb/main.cpp [hunk 1]
```diff
@@ -121,6 +121,7 @@ static void setApplicationMetadata() noexcept
     Application::setOrganizationName("LibrePCB");
     Application::setOrganizationDomain("librepcb.org");
     Application::setApplicationName("LibrePCB");
+    Application::setDesktopFileName("org.librepcb.LibrePCB");
```
[the remark this answers is not in the document]
Comment (Alexander-Wilms) [reply apps/librepcb/main.cpp]: Thanks for the info, done

**Comment by ubruhin** [comment]
I added the commit 841ab830a43a9b61c0923a548b5295c718a5222c to fix missing images when installed via the installer. Now it seems to work properly.

Just one thing: Do you know what the `Application::setDesktopFileName()` is actually used for? I ask because the filename "org.librepcb.LibrePCB" is not correct when LibrePCB is installed with the installer, and I don't know if this could break something or not. In the installer I renamed the desktop file to "librepcb-nightly-from-installer.desktop" to avoid name conflicts if LibrePCB is already installed via package manager. 

Actually for me it feels bad to compile the name of the desktop file into the application because at the time of compilation it may not be clear yet how it is deployed afterwards, and thus how the desktop file will be named.

**Comment by Alexander-Wilms** [comment]
It's somehow used by the windowmanager to associate the desktop file with the window, but I don't know how it actually works.

**Comment by ubruhin** [comment]
OK let's go with this for the moment, we can still fix it later if we encounter issues with the installer deployments.

Thanks for your help @Alexander-Wilms :smiley:

**Comment by Alexander-Wilms** [comment]
Thanks for merging :+1:

**Pull request merged**